File: http/src/main/Plume/Http/Routes/Server/DeleteRoute.kt
package Plume.Http.Routes.Server

import Plume.Configuration.ConfigManager
import Plume.Server.Manager
import Plume.Http.RouteModule
import Plume.Http.Routes.errorJson
import Plume.Http.Routes.successJson
import io.ktor.server.request.receiveText
import io.ktor.server.response.respond
import io.ktor.server.routing.Route
import io.ktor.server.routing.post
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.jsonObject
import kotlinx.serialization.json.jsonPrimitive

object DeleteRoute : RouteModule {
    override fun register(route: Route) {

        route.post("/servers/delete") {
            val body = call.receiveText()
            val json = Json.parseToJsonElement(body).jsonObject

            val serverId = json["serverId"]?.jsonPrimitive?.content
            val userUuid = json["userUuid"]?.jsonPrimitive?.content

            if (serverId == null || userUuid == null) {
                call.respond(errorJson("Missing required fields: serverId and userUuid"))
                return@post
            }

            if(!ConfigManager.validateServerID(serverId)) {
                call.respond(errorJson("Invalid serverId"))
                return@post
            }

            if(!ConfigManager.validateUUID(userUuid)) {
                call.respond(errorJson("Invalid userUuid"))
                return@post
            }

            if(!ConfigManager.userIsAdmin(userUuid)) {
                call.respond(errorJson("User does not have admin permissions"))
                return@post
            }

            if(Manager.get(serverId) == null) {
                call.respond(errorJson("Server not found"))
            }
            Manager.delete(serverId)
            call.respond(successJson())
        }
    }
}

File: http/src/Plume/Routes/Server/DeleteRoute.cs
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Routing;
using Plume.Configuration;
using Plume.Server;

namespace Plume.Http.Routes.Server
{
    public static class DeleteRoute
    {
        // DTO para mapear o JSON recebido no body
        public record DeletePayload(
            string ServerId = "",
            string UserUuid = ""
        );

        public static void Register(RouteGroupBuilder group)
        {
            group.MapPost("/servers/delete", (DeletePayload body) =>
            {
                // Verifica se os campos obrigatórios vieram nulos ou vazios
                if (string.IsNullOrEmpty(body.ServerId) || string.IsNullOrEmpty(body.UserUuid))
                    return Results.Json(new { error = "Missing required fields: serverId and userUuid" }, statusCode: 400);

                // Validação do Server ID
                if (!ConfigManager.ValidateServerID(body.ServerId))
                    return Results.Json(new { error = "Invalid serverId" }, statusCode: 400);

                // Validação do UUID do usuário
                if (!ConfigManager.ValidateUUID(body.UserUuid))
                    return Results.Json(new { error = "Invalid userUuid" }, statusCode: 400);

                // Validação de permissões de administrador
                if (!ConfigManager.UserIsAdmin(body.UserUuid))
                    return Results.Json(new { error = "User does not have admin permissions" }, statusCode: 403);

                // Verifica se o servidor existe no Manager
                if (Manager.Get(body.ServerId) == null)
                    return Results.Json(new { error = "Server not found" }, statusCode: 404);

                // Deleta o servidor usando o Manager
                Manager.Delete(body.ServerId);

                // Retorna o JSON de sucesso
                return Results.Ok(new { status = "success" });
            });
        }
    }
}